tests/5.1/target_update/test_target_update_iterator.F90

```diff
@@ -0,0 +1,93 @@
+!===--- test_target_update_iterator.F90    -----------------------------===//
+!
+!  OpenMP API Version 5.1 Aug 2021
+!
+!  This test uses the target update directive with the iterator clause
+!  to update the iterative values.
+!
+!  This example has been adapted from the 5.2 OpenMP Examples document,
+!  "Multidependences Using Iterators" and "Simple target data and target update
+!  constructs"
+!
+!===-----------------------------------------------------------------------===//
+
+#include "ompvv.F90"
+
+#define N 1024
+
+MODULE my_struct
+  TYPE test_struct
+    INTEGER :: len
+    INTEGER, POINTER:: data(:)
+  END TYPE
+END MODULE
+
+PROGRAM main
+    USE my_struct
+    USE iso_fortran_env
+    USE ompvv_lib
+    USE omp_lib
+    implicit none
+    
+    OMPVV_TEST_OFFLOADING
+
+    OMPVV_TEST_VERBOSE(test_target_update_iterator() .NE. 0)
+
+    OMPVV_REPORT_AND_RETURN()
+
+CONTAINS
+    SUBROUTINE init(s)
+        TYPE(test_struct), INTENT(IN), POINTER :: s
+        INTEGER :: i
+        s%len = N
+        ALLOCATE(s%data(N))
+        DO i = 1, s%len
+            s%data(i) = i
+        END DO
+    END SUBROUTINE
+
+    SUBROUTINE init_again(s)
+        TYPE(test_struct), INTENT(IN), POINTER :: s
+        INTEGER :: i
+        s%len = N
+        ALLOCATE(s%data(N))
```

Comment:
is it necessary to allocate data again in this subrutine?, given that the `s` (`new_struct`) was allocated in `init`?

Reply:
You are correct; we don't need to reallocate the data.